go-controller/hybrid-overlay/pkg/controller/node_linux.go

```diff
@@ -353,8 +353,156 @@ func getIPAsHexString(ip net.IP) string {
 	return asHex
 }
 
+// swaps out the new values of drIP and drMAC
+func (n *NodeController) recalculateFlowCache(oldDRIP net.IP, oldDRMAC net.HardwareAddr) {
+	n.flowMutex.Lock()
+	defer n.flowMutex.Unlock()
+
+	var ipReplace, macReplace int
+
+	if oldDRIP != nil {
+		ipReplace = -1
+	}
+	if oldDRMAC != nil {
+		macReplace = -1
+	}
+	oldDRMACRaw := strings.Replace(oldDRMAC.String(), ":", "", -1)
+	newDRMACRaw := strings.Replace(n.drMAC.String(), ":", "", -1)
+
+	portIPRawOld := getIPAsHexString(oldDRIP)
+	portIPRawNew := getIPAsHexString(n.drIP)
+
+	for _, entry := range n.flowCache {
+		for i, flow := range entry.flows {
+			replacementFlow := flow
+			replacementFlow = strings.Replace(replacementFlow, oldDRIP.String(), n.drIP.String(), ipReplace)
+			replacementFlow = strings.Replace(replacementFlow, portIPRawOld, portIPRawNew, ipReplace)
+
+			replacementFlow = strings.Replace(replacementFlow, oldDRMACRaw, newDRMACRaw, macReplace)
+			replacementFlow = strings.Replace(replacementFlow, oldDRMAC.String(), n.drMAC.String(), macReplace)
+			entry.flows[i] = replacementFlow
+
+		}
+	}
+}
+
+func (n *NodeController) createOrReplaceRoutes(mgmtPortLink netlink.Link, oldDRIP net.IP) error {
+	// hybridOverlay has been initialized and will needs to delete the old routes
+	if oldDRIP != nil {
+		if len(config.HybridOverlay.ClusterSubnets) > 0 {
+			for _, clusterEntry := range config.HybridOverlay.ClusterSubnets {
+				route := &netlink.Route{
+					Dst:       clusterEntry.CIDR,
+					LinkIndex: mgmtPortLink.Attrs().Index,
+					Scope:     netlink.SCOPE_UNIVERSE,
+					Gw:        oldDRIP,
+				}
+				err := util.GetNetLinkOps().RouteDel(route)
+				if err != nil && !os.IsExist(err) {
+					return fmt.Errorf("failed to add route for subnet %s via gateway %s: %v",
+						route.Dst, route.Gw, err)
+				}
+			}
+		} else {
+			nodes, err := n.nodeLister.List(labels.Everything())
+			if err != nil {
+				return err
+			}
+			for _, node := range nodes {
+				if subnet, _ := houtil.ParseHybridOverlayHostSubnet(node); subnet != nil {
+					route := &netlink.Route{
+						Dst:       subnet,
+						LinkIndex: mgmtPortLink.Attrs().Index,
+						Scope:     netlink.SCOPE_UNIVERSE,
+						Gw:        oldDRIP,
+					}
+					err := util.GetNetLinkOps().RouteDel(route)
+					if err != nil && !os.IsExist(err) {
+						return fmt.Errorf("failed to add route for subnet %s via gateway %s: %v",
+							route.Dst, route.Gw, err)
+					}
+				}
+			}
+		}
+	}
+	// Add a route via the hybrid overlay port IP through the management port
+	// interface for each hybrid overlay cluster subnet
+	if len(config.HybridOverlay.ClusterSubnets) > 0 {
+		for _, clusterEntry := range config.HybridOverlay.ClusterSubnets {
+			route := &netlink.Route{
```

Comment:
Same here, we could pull this out to a function to add the route that both blocks could use.

Reply:
And also use in `hybridOverlayNodeUpdate()`

Reply:
done